runbooks/mcp-l1.md

```diff
@@ -0,0 +1,352 @@
+CLI tool from the monorepo to generate the `input.json`. (The README there may be outdated due to
+some changes we made, but this guide will contain all the info you need).
+
+Below is the syntax we’ll use to run this tool from the root of the monorepo:
+
+```bash
+SYSTEM_CONFIG_START_BLOCK={startBlock} go run op-chain-ops/cmd/op-upgrade/main.go \
+  --l1-rpc-url [mainnet or sepolia L1 RPC URL] \
+  --chain-ids [chain ID of the L2 to upgrade] \
+  --superchain-target [name of the superchain target to upgrade, either mainnet or sepolia] \
+  --outfile input.json
+```
+
+The `startBlock` is the the block number at which the `SystemConfig` proxy was initialized for the
+first time. To find this value, find the very first `Initialize` event after the `SystemConfig`
+was deployed. One way to do this is view the Etherscan events page for the contract. This only
+shows the last 25 or less events, which is often enough. Etherscan will tell you the transaction
+hash (and therefore block) in which the contract was deployed. If the oldest event that you see on
+the page is that deployment block, and you only see one `Initialize` event, then you can use the
+block number of the `Initialize` event in the UI. An alternate approach is to use `cast logs` to
+fetch all `Initialize` events and find the oldest.
+
+The other parameters are self-explanatory given the above descriptions. A sample invocation for Metal Sepolia looks like this:
+
+```bash
+SYSTEM_CONFIG_START_BLOCK=5304055 go run op-chain-ops/cmd/op-upgrade/main.go \
+  --l1-rpc-url $SEPOLIA_RPC_URL \
+  --chain-ids 1740 \
+  --superchain-target sepolia \
+  --outfile input.json
+```
+
+A successful execution will generate the `input.json` file.
+
+### Simulate and Validate
+
+> [!WARNING]
+> This is part is the most complicated part of the process, so its recommended to do this while
+> you’re well rested and can carefully validate the transaction simulation results.
+
+Copy the `input.json` from the monorepo into the task you created in the ops repo. It should live
+at `tasks/{NetworkDir}/{chainName}-001-MCP-L1/input.json`
+
+Next, open the `SignFromJson.s.sol` in your task. This has various hardcoded values that you may
+need to change, including:
+
+- The `l1ChainName` and `l2ChainName` values.
+- The 6 addresses under the `// Known EOAs to exclude from safety checks.` comment.
+  - These have comments explaining where to get each value from.
+  - Context on these addresses: Most upgrades have an an invariant of “When writing an address into a storage slot, that address should *usually* have code”. It’s “usually” and not “always” because:
+    - Some roles in the protocol are EOAs with no code, and their addresses are in storage. This is what these 6 EOAs are. Depending on the chain configuration, this list of six addresses may be too large or too small, so edit it as appropriate.
```

Comment:
Is it worth saying you may have to manually edit the `getCodeExceptions` function in the `SignFromJson.s.sol` file? e.g. removing an address that has code at it when you expect there to be i.e. gnosis safe instead of an EOA.

Reply:
Done in e0bccf0